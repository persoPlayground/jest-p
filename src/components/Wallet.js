import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setBalance, addDepeosit,  withdraw } from '../actions/balance';
export class Wallet extends Component {

    state = {
        balance: undefined
    }

    updateBalance = e =>{
        this.setState({balance: parseInt(e.target.value, 10)});
    }
    deposit = () => this.props.addDepeosit(this.state.balance);
    remove  = () => this.props.withdraw(this.state.balance);
    render(){
        return(
            <div>
                <h3 className="balance">Wallet Balance: { this.props.balance }</h3>
            
            <br />
            <input className="input-wallet" onChange={this.updateBalance} />
            <button className="btn-deposit" onClick={this.deposit}>Deposit</button>
            <button className="btn-remove" onClick={this.remove}>withdraw</button>
            </div>
        )
    }
}

export default connect(state => ({balance: state}),{setBalance, addDepeosit, withdraw })(Wallet);