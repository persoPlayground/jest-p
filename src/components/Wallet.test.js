import React from 'react';
import { shallow } from 'enzyme';
import {Wallet} from './Wallet';

describe('Wallet', () => {
const mockDeposit = jest.fn();  
const mockWithdraw =jest.fn(); 
const props = {
    balance: 30,
    withdraw : mockWithdraw,
    addDepeosit: mockDeposit
}
const wallet = shallow(<Wallet {...props} />);

it('renders correctly', () => {
    expect(wallet).toMatchSnapshot();
});

it('displays Balance from props', () =>{
    expect(wallet.find('.balance').text()).toEqual('Wallet Balance: 30');
});

describe('Add deposit to wallet', () =>{
it('creates an input to deposit', () =>{
    expect(wallet.find('.input-wallet').exists()).toBe(true);
});

describe('updates the state when type deposit', () => {
    const userBalance = '50';
    beforeEach(() => {
        wallet.find('.input-wallet').simulate('change',{
            target: { value: userBalance}
        })
    });
   it('changes the local balance in state and convert it to Number', ()=>{
    expect(wallet.state().balance).toEqual(parseInt(userBalance, 10));
   }); 

   describe('and user wants to make a deposit', () =>{

    beforeEach(() => {
        wallet.find('.btn-deposit').simulate('click');
    });
    it('dispatch deposit action `deposit()` received in props with the local balance', () =>{
       expect(mockDeposit).toHaveBeenCalledWith(parseInt(userBalance, 10)); 
    });
});


describe('remove deposit from wallet', () => {

    beforeEach(() => {
        wallet.find('.btn-remove').simulate('click');
    });

    it('dispatchs the withdraw() receuved from the local state', () => {
        expect(mockWithdraw).toHaveBeenCalledWith(parseInt(userBalance, 10));
    });


});

});



});



});
