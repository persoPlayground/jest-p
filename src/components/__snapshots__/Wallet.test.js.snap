// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    addDepeosit={[Function]}
    balance={30}
    withdraw={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="balance"
>
          Wallet Balance: 
          30
</h3>,
        <br />,
        <input
          className="input-wallet"
          onChange={[Function]}
/>,
        <button
          className="btn-deposit"
          onClick={[Function]}
>
          Deposit
</button>,
        <button
          className="btn-remove"
          onClick={[Function]}
>
          withdraw
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Wallet Balance: ",
            30,
          ],
          "className": "balance",
        },
        "ref": null,
        "rendered": Array [
          "Wallet Balance: ",
          30,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-wallet",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Deposit",
          "className": "btn-deposit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Deposit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "withdraw",
          "className": "btn-remove",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "withdraw",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="balance"
>
            Wallet Balance: 
            30
</h3>,
          <br />,
          <input
            className="input-wallet"
            onChange={[Function]}
/>,
          <button
            className="btn-deposit"
            onClick={[Function]}
>
            Deposit
</button>,
          <button
            className="btn-remove"
            onClick={[Function]}
>
            withdraw
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Wallet Balance: ",
              30,
            ],
            "className": "balance",
          },
          "ref": null,
          "rendered": Array [
            "Wallet Balance: ",
            30,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-wallet",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Deposit",
            "className": "btn-deposit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Deposit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "withdraw",
            "className": "btn-remove",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "withdraw",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
